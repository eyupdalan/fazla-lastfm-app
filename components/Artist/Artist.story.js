import Artist from "./index";

const artistStory = {
    title: 'Components/Artist',
    component: Artist,
};

export const ArtistStory = () => <Artist/>;
export const ArtistStoryWithData = () => (
    <Artist
        mbid={"test"}
        name={"Linkin Park"}
        listeners={5000}
        playCount={10000}
        image={"https://upload.wikimedia.org/wikipedia/commons/4/42/LinkinParkBerlin2010.jpg"}
    />
);

export default artistStory;