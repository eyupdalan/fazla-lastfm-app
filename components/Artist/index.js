import PropTypes from "prop-types";
import styles from "../../styles/Artist.module.css";
import Image from "../Image";
import ArtistName from "../ArtistName";

export default function Artist({mbid, name, image, listeners, playCount}) {
    return (
        <div className={styles.artist}>
            <div className={styles.artistImage}>
                <Image src={image} alt={name} height={100} width={100}/>
            </div>
            <div className={styles.artistDetailContainer}>
                <h5>Artist</h5>
                <div className={styles.artistDetail}>
                    <ArtistName mbid={mbid} name={name}/>
                    <div>
                        <div>Listeners: {listeners}</div>
                        <div>Play count: {playCount}</div>
                    </div>
                </div>
            </div>
        </div>
    )
}

Artist.propTypes = {
    mbid: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    listeners: PropTypes.number.isRequired,
    playCount: PropTypes.number.isRequired
}

Artist.defaultProps = {
    mbid: "",
    name: "",
    image: "",
    listeners: 0,
    playCount: 0
}