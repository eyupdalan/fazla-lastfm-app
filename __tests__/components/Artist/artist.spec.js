import { render } from '@testing-library/react'
import Artist from '../../../components/Artist';

it('renders Artist component', () => {
    const { container } = render(<Artist />)
    expect(container).toMatchSnapshot()
})
