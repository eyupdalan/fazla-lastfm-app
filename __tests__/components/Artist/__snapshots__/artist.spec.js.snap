// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Artist component 1`] = `
<div>
  <div
    class="artist"
  >
    <div
      class="artistImage"
    >
      <div>
        No Image
      </div>
    </div>
    <div
      class="artistDetailContainer"
    >
      <h5>
        Artist
      </h5>
      <div
        class="artistDetail"
      >
        <div
          class="artistName"
        >
          <a
            href="/artist"
          />
        </div>
        <div>
          <div>
            Listeners: 
            0
          </div>
          <div>
            Play count: 
            0
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
